### PR TITLE
Naxuroqa upgrade circleci config

### PR DESCRIPTION
**Pull request by sudden6**
- [ ] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

continuation of #5504

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5548)

### CODE REVIEW

**Comment by naxuroqa** [comment]
Maybe the caching can still be improved, circleci offers some quite complex caching strategies. Otherwise LGTM :+1:

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+version: 2.1
+jobs:
+  build:
```
Comment (naxuroqa) [inline .circleci/config.yml:4]: Just a heads up, it doesn't need to be named "build" - you can give it an easily identifyable name, which will also show up as this on github. If you are interested in adding more jobs on circleci.

See e.g. https://github.com/naxuroqa/Venom/blob/develop/.circleci/config.yml#L103

Comment (sudden6) [reply .circleci/config.yml:4]: Thanks for the hints, I think I'll consult with you before I make additional updates to our CircleCI system.

**Comment by sudden6** [comment]
@naxuroqa could you give the :lgtm_strong: via reviewable? then I could merge it.

**Review by naxuroqa — COMMENTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5548)* status: 0 of 1 LGTMs obtained (waiting on @anthonybilinski)

**Comment by naxuroqa** [comment]
:+1: though I can't officially approve it, as I'm not a member of qtox

**Review by sudden6 — COMMENTED** [review]
Since @anthonybilinksi seems inactive, lets see if i can :lgtm_strong: this myself.

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5548#-:-LZ5qY2R-8aIU7bWUdzZ:bnden0i)* status: :shipit: complete! 1 of 1 LGTMs obtained

**Pull request merged**